aspnetcore/client-side/blazor/introduction/index.md

```diff
@@ -0,0 +1,130 @@
+* Full-stack development that leverages existing skills: C#/Razor developers use their existing C#/Razor skills to write client-side code and share server and client-side logic among apps.
+* Wide browser support: Blazor runs on .NET using open web standards in the browser with no plugins and no code transpilation. It works in all modern web browsers, including mobile browsers.
+
+## Running .NET in the browser
+
+Running .NET code inside web browsers is made possible by a relatively new technology, WebAssembly (abbreviated *wasm*). WebAssembly is an open web standard and is supported in web browsers without plugins. WebAssembly is a compact bytecode format optimized for fast download and maximum execution speed.
+
+Security isn't a major concern because WebAssembly isn't ordinary assembly code (for example, x86/x64) &mdash; WebAssembly is a new bytecode format that accesses browser functionality with the same capabilities as JavaScript.
+
+When a Blazor app is built and run in a browser:
+
+1. C# code files and Razor files are compiled into .NET assemblies.
+1. The assemblies and the .NET runtime are downloaded to the browser.
+1. Blazor uses JavaScript to bootstrap the .NET runtime and configures the runtime to load required assembly references. Document object model (DOM) manipulation and browser API calls are handled by the Blazor runtime via JavaScript interoperability.
+
+## Browsers that don't support WebAssembly
+
+The .NET runtime is supplied as a WebAssembly binary and an [asm.js](https://en.wikipedia.org/wiki/Asm.js)-based implementation. *asm.js* is a subset of JavaScript and can be executed by JavaScript runtimes in browsers going back several years. When Blazor loads in the browser, it checks for WebAssembly support. If WebAssembly isn't supported, the *asm.js* runtime is loaded. *asm.js* isn't always used because it's larger and slower than the WebAssembly runtime.
+
+## Blazor components
+
+In client-side web UI frameworks, apps are built with *components*. A component usually represents a piece of UI, such as a page, dialog, or data entry form. Components can be nested, reused, and shared between projects.
+
+In Blazor, a component is a .NET class. The class can either be written directly, as a C# class (*\*.cs*), or more commonly in the form of a Razor markup page (*\*.cshtml*).
+
+Many design patterns are possible using [Razor](xref:mvc/views/razor) as a foundation for Blazor. Razor is a syntax for combining HTML markup with C# code. Razor is designed for developer productivity, allowing the developer to switch between markup and C# in the same file with IntelliSense support. The following markup is an example of a basic custom dialog component in a Razor file:
+
+```cshtml
+<div>
+    <h2>@Title</h2>
+    @RenderContent(Body)
+    <button onclick=@OnOK>OK</button>
+</div>
+
+@functions {
```

Comment:
@danroth27 We wanted to discourage the use of the `@functions` directive in the Razor Pages docs. Should we avoid using it here too?

Reply:
I need general guidance on this, too. Samples and blog posts include them, and all of the talks on Blazor seem to feature this approach. Shall I continue in that vein as I go forward with examples and the tutorial?